src/main.cpp

```diff
@@ -920,6 +923,46 @@ struct router_thread
 			|| cmd == "get_peers"
 			|| cmd == "get")
 		{
+			insert_response inserted = insert_response::inserted;
```

Comment:
I flipped around sending the response and inserting into the queue, since inserting is what tells us whether it's a duplicate or not. note that I removed your duplicate check against the last source IP. did you get a lot of hits on that one? should I put it back?